src/movies.js

```diff
@@ -1,16 +1,79 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    return directorsArray
+}
+
+const directorsArray = movies.map ((element) => {
+    return element.director;
+});
+
+getAllDirectors(movies) 
+
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  let count = 0;
+
+const moviesSpielberg = moviesArray.filter ((element) => {
+  if (element.director === 'Steven Spielberg') {
+    count ++;
+    return true;
+    }
+});
+    
+}
+
+howManyMovies(movies)
+
+
+
+    
+
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
```

Comment:
Here it would be something similar to the first one, you probably should have written your code inside the function and you would need do some checking, like for and empty array and if the object has the `currentValue.score`

A possible solution:

```
function scoresAverage(moviesArray) {
	if (moviesArray.length === 0) {
		return 0
	}

	const scores = moviesArray.reduce(function (accumulator, currentValue) {
		if (typeof currentValue.score !== 'number') {
			currentValue.score = 0
		}

		return accumulator + currentValue.score
	}, 0)

	const averageResult = scores / moviesArray.length
	const decimals = +averageResult.toFixed(2)
	console.log(decimals)
	return decimals
}
```